xbmc/application/ApplicationPlayerCallback.cpp

```diff
@@ -245,11 +245,8 @@ void CApplicationPlayerCallback::OnPlayBackSeek(int64_t iTime, int64_t seekOffse
   param["player"]["speed"] = (int)m_appPlayer.GetPlaySpeed();
   CServiceBroker::GetAnnouncementManager()->Announce(ANNOUNCEMENT::Player, "OnSeek",
                                                      m_itemCurrentFile, param);
-  CServiceBroker::GetGUI()
-      ->GetInfoManager()
-      .GetInfoProviders()
-      .GetPlayerInfoProvider()
-      .SetDisplayAfterSeek(2500, static_cast<int>(seekOffset));
+
+  CDataCacheCore::GetInstance().SeekFinished(static_cast<int>(seekOffset));
```

Comment:
The use of `GetInstance()` means that CDataCacheCore is a singleton, which we are moving away from because singletons don't give you control of construction/destruction lifetime functions. A good follow-up for anyone interested would be removing this singleton.

Reply:
Yeah I know, thanks for noticing that. The class "needs" to be a singleton (or at least unique) since it caches state from several components - acting like a cache repo for GUI/info to access. Doesn't seem difficult to kill though as there are no dependencies (only users) of the class. Maybe we can have some sort of cache component in the broker which will serve as parent of datacachecore and other similar instances.
Not something added with this PR, was already a singleton (and used by a singleton) by several classes.
Will try to have a look later

Reply:
Because this is an existing issue it's definitely not something for this PR.